### PR TITLE
Move custom reports config to prepend extension

### LINKED ISSUE #398: [Task]: Adapt config.yaml for custom report adapter
### Improvement description

Adapt config.yaml after https://github.com/pimcore/pimcore/issues/12516 gets merged

### PR DESCRIPTION
**Pull request by dvesh3**
Resolves #398

### CODE REVIEW

File: src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php [hunk 1]
```diff
@@ -39,14 +40,20 @@ class PimcoreCustomerManagementFrameworkExtension extends ConfigurableExtension
...
+            new FileLocator(__DIR__ . '/../Resources/config')
+        );
 
+        if ($container->hasExtension('doctrine_migrations')) {
             $loader->load('doctrine_migrations.yml');
         }
+
+        if ($container->hasExtension('pimcore_custom_reports')) {
+            $loader->load('custom_reports.yml');
+        } elseif (Version::getMajorVersion() < 11) { //@TODO BC layer for Pimcore 10
```
Comment (kingjia90) [inline src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php]:
```suggestion
        } elseif (Version::getMajorVersion() < 11) { //@TODO remove BC layer when dropping support for Pimcore 10
```

File: src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php [hunk 2]
```diff
@@ -39,14 +40,20 @@ class PimcoreCustomerManagementFrameworkExtension extends ConfigurableExtension
...
+            $container,
+            new FileLocator(__DIR__ . '/../Resources/config')
+        );
 
+        if ($container->hasExtension('doctrine_migrations')) {
             $loader->load('doctrine_migrations.yml');
         }
+
+        if ($container->hasExtension('pimcore_custom_reports')) {
+            $loader->load('custom_reports.yml');
```
Comment (kingjia90) [inline src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php]:
```suggestion
            $loader->load('custom_reports.yaml');
```
just wondering, should we start recommending `yaml` for 11 related stuff?

File: src/Resources/config/custom_reports_legacy.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+services:
+    _defaults:
+      public: true
+      autowire: true
+      autoconfigure: true
+
+    cmf.custom_report.adapter.term_segment_builder:
+      class: Pimcore\Model\Tool\CustomReport\Adapter\DefaultCustomReportAdapterFactory
+      arguments:
+      - 'CustomerManagementFrameworkBundle\CustomReport\Adapter\TermSegmentBuilder'
```
Comment (kingjia90) [inline src/Resources/config/custom_reports_legacy.yml]:
```suggestion
          - 'CustomerManagementFrameworkBundle\CustomReport\Adapter\TermSegmentBuilder'
```

File: src/CustomReport/Adapter/TermSegmentBuilderBase.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace CustomerManagementFrameworkBundle\CustomReport\Adapter;
+
+use Pimcore\Bundle\CustomReportsBundle\Tool\Adapter\Sql;
+use Pimcore\Version;
+
+//@TODO BC layer for Pimcore 10
```
Comment (kingjia90) [inline src/CustomReport/Adapter/TermSegmentBuilderBase.php]:
```suggestion
//@TODO remove BC layer when dropping support for Pimcore 10
```

**Review by kingjia90 — CHANGES_REQUESTED** [review]
LGTM, just some minor feedback

**Comment by kingjia90** [comment]
https://github.com/pimcore/customer-data-framework/blob/15146baf4efabc470804e8d218d6166d5439b537/src/Resources/public/js/pimcore/report/custom/definitions/termSegmentBuilder.js#L15-L16

In 11, it needs to be changed to `pimcore.bundle.customreports.custom.definition.termSegmentBuilder` or it won't appear in the [Source Definition list](https://github.com/pimcore/pimcore/pull/13920/files?w=1#diff-fcf58cd506dd39ccc7be87256a1a204433e2fa462dfcb9517f8e5c1bfff83164L780). eg. [analytics](https://github.com/pimcore/pimcore/pull/13920/files?w=1#diff-d3f6942a34843d69fa64b207d286ad004b4daa1deff53a01322932c06f649d02)
The rest looks fine to me.

**Comment by dvesh3** [comment]
> https://github.com/pimcore/customer-data-framework/blob/15146baf4efabc470804e8d218d6166d5439b537/src/Resources/public/js/pimcore/report/custom/definitions/termSegmentBuilder.js#L15-L16
> 
> In 11, it needs to be changed to `pimcore.bundle.customreports.custom.definition.termSegmentBuilder` or it won't appear in the [Source Definition list](https://github.com/pimcore/pimcore/pull/13920/files?w=1#diff-fcf58cd506dd39ccc7be87256a1a204433e2fa462dfcb9517f8e5c1bfff83164L780). eg. [analytics](https://github.com/pimcore/pimcore/pull/13920/files?w=1#diff-d3f6942a34843d69fa64b207d286ad004b4daa1deff53a01322932c06f649d02) The rest looks fine to me.

Done

**Review by kingjia90 — APPROVED** [review]

**Pull request merged**